tasmota/xnrg_15_teleinfo.ino

```diff
@@ -324,6 +319,62 @@ void NewFrameCallback(struct _ValueList * me)
 {
     // Reset Energy Watchdog
     Energy.data_valid[0] = 0;
+
+    // send teleinfo full frame only if setup like that
+    // see setOption108
+    if (Settings.flag4.teleinfo_rawdata) { 
+        struct _ValueList * me = tinfo.getList();
+
+        // Total size we can use, remove { } and \0
+        int remaining_size;
+        char sep = ' ';
+        char * p ;
+        int len;
+        boolean isNumber ;
+
+        strcpy_P(mqtt_data, PSTR("{"));
+        len = strlen(mqtt_data);
```

Comment:
Dive into the use of Response function. This increases code size with redundant code.

Reply:
Theo,

Not sure to understand, the deal is that code happens on frame reception so mainly called in 250ms function not in Web or JSON answer.
just followed the https://tasmota.github.io/docs/Sensor-API/ MQTT part, now if a better solution exists I'll take it but not sure what to do.

Reply:
Any manipulation of mqtt_data should be done using the provided respond functions. Search code and it wil become clear. No need for strcat or printf

Reply:
Ok, got it, just followed documentation should have dig into the code, I'll fix that tomorrow.

```
      Response_P(PSTR("{"));
      ResponseAppend_P( bla bla );
      ResponseJsonEnd();
```

Reply:
Theo,
That's done